Appleton: Initial Commit

onuro opened the pull request:
This is the initial commit of the Appleton (Blockbase child) theme. This theme is a part of the verticals project (more info: pbxlJb-18I-p2).

My first premium theme, so I might be doing some things wrong :) Please correct me where I need to!

Thanks!

Homepage Example and [Draft Demo →](https://appleton105180900.wordpress.com/)

![Appleton - Photographer](https://user-images.githubusercontent.com/645713/165540587-faa98ec1-a62c-494e-85e4-933d83536bee.jpg)

onuro committed:
Initial Commit

Appleton - initial commit

pbking commented on appleton/parts/header.html:
```diff
@@ -0,0 +1,17 @@
+<!-- /wp:group --></div>
+<!-- /wp:group -->
+
+<!-- wp:navigation {"ref":22,"overlayMenu":"always","__unstableLocation":"primary","layout":{"type":"flex","setCascadingProperties":true,"justifyContent":"right","orientation":"horizontal"},"style":{"spacing":{"margin":{"top":"0"}}}} /--></div>
```
The navigation block template part shouldn't have a ref id.  Otherwise the `__unstableLocation` bits won't work.

pbking commented:
The demo site looks smashing!  This theme will be really good when it matches.

A few things to point out about making it so:

The font weights represented in your demo don't seem to be present in theme.json (or the templates)

(my site)
<img width="865" alt="image" src="https://user-images.githubusercontent.com/146530/165964304-51db7b6c-ad2e-4623-99a9-94c4251949e5.png">

For instance that `site-title` defaults to 700 weight but in your example site is 300.  That should be defined either in the theme.json (`{style.blocks.core/site-title.typography.fontWeight}`) for all instances of the site-title block or in the header template to effect only that one.

The same is true for font weights which are defaulting to 400 but your demo is lighter.

The footer in the demo looks really good, but should be represented in a template-part or pattern.

onuro committed:
Template Parts Update

Added custom header and footer template parts w styling based on demo.

onuro commented:
@pbking thank you for the feedback! I've updated the typography settings rather on the template compared to site-wide weights since the layout is mostly built via custom patterns made for this theme. 

But if you think I should still apply those weights site-wide, I can do that too. Please let me know.

Also updated header/footer templates.

onuro committed:
Header template ref id removal

And theme.json version value update to 1.

pbking commented:
@onuro I would suggest changing as much as you can at the theme level (meaning, in the theme.json configuration) rather than the pattern level.  In this way the patterns you create for this theme will be more likely to fit in any site's design (and likewise, other patterns more likely to fit into this theme's design).

If I create content (even just a post with headers and paragraphs) the result is noticeably different than if I exclusively use the patterns included.

onuro committed:
Typography and color updates

Typography updates in theme.json and header template padding value matching with the demo

onuro commented:
Thanks @pbking! I've updated the values as much as I can globally. I wanted to add uppercase to all headings but looks like it's not working for a headings setting yet.

Let me know if this works.

Thanks!

mikachan commented on appleton/parts/header.html:
```diff
@@ -0,0 +1,17 @@
+<!-- /wp:group --></div>
+<!-- /wp:group -->
+
+<!-- wp:navigation {"ref":22,"overlayMenu":"always","__unstableLocation":"primary","layout":{"type":"flex","setCascadingProperties":true,"justifyContent":"right","orientation":"horizontal"},"style":{"spacing":{"margin":{"top":"0"}}}} /--></div>
```
It looks like the ref id still needs to be removed here:

```suggestion
<!-- wp:navigation {"overlayMenu":"always","__unstableLocation":"primary","layout":{"type":"flex","setCascadingProperties":true,"justifyContent":"right","orientation":"horizontal"},"style":{"spacing":{"margin":{"top":"0"}}}} /--></div>
```

onuro replied:
Yeah sorry I missed it after the second export. Fixed now.

mikachan commented on appleton/parts/footer.html:
```diff
@@ -0,0 +1,39 @@
+<div class="wp-block-group"><!-- wp:site-title {"style":{"elements":{"link":{"color":{"text":"var:preset|color|background"}}},"typography":{"fontStyle":"normal","fontWeight":"600","fontSize":"1.2rem","textTransform":"uppercase"}},"textColor":"background"} /-->
+
+<!-- wp:paragraph -->
+<p>Photography portfolio website of a nature photographer,&nbsp;<strong>John Appleton</strong>.</p>
```
These strings should be translatable. You need to use PHP for this, so I'd recommend creating a pattern (PHP file) for this footer, and then you can wrap the strings with `esc_html_e`, like this: `<?php esc_html_e( 'Text goes here.', 'appleton' ); ?>`

Example pattern in Archeo: https://github.com/Automattic/themes/blob/trunk/archeo/patterns/hidden-404.php#L14
Here's the pattern being used: https://github.com/Automattic/themes/blob/trunk/archeo/templates/404.html#L6

This `footer.html` would then just contain something like this, similar to Archeo's [footer](https://github.com/Automattic/themes/blob/trunk/archeo/parts/footer.html):

```
<!-- wp:pattern {"slug":"appleton/footer"} /-->
```

onuro replied:
Thanks! updated according to reference theme.

mikachan reviewed (commented):
This is looking good! I've left a couple of comments below.

Also, the screenshot has a pretty large file size at 1.25MB. Screenshot files are usually around or under 100KB. I don't think there's a rule for this, but I think it would be good if you could compress this some more.

onuro committed:
Template parts and Screenshot updates

- Removed ref from the header
- Footer as a template part
- Compressed Screenshot.png

mikachan reviewed (approved):
Thanks, those changes look great! I think this is ready to bring in 🎉
